src/door-status.ts

```diff
@@ -0,0 +1,64 @@
+import { EmbedBuilder, Message, TextChannel } from "discord.js";
+import persist from "node-persist";
+
+// Key to store the last message ID in local storage
+const storage_key = "door_status";
+
+let last_message_id: string | null = null;
+
+export async function initializeDoorStatusMessage(channel: TextChannel) {
+    // Fetch the last message ID from local storage
+    const existing_message = (await persist.getItem(storage_key)) as Message | null;
+    if (existing_message) {
+        // Fetch the message using the stored ID
+        const message = await channel.messages.fetch(existing_message.id);
+        if (message) {
+            // Store the fetched message's ID
+            last_message_id = message.id;
+        }
+    }
+}
+
+export async function updateDoorStatusMessage(channel: TextChannel, state: boolean) {
+    try {
+        // Check if there's a previous message
+        if (last_message_id) {
+            // Fetch the previous message using the stored ID
+            const existing_message = await channel.messages.fetch(last_message_id);
+            if (existing_message) {
+                // Delete the previous message
+                await existing_message.delete();
+            }
+        }
+
+        // Create an embed with the updated door status
+        const embed = createDoorStatusEmbed(state);
+
+        // Send the embed as a new message
+        const message = await channel.send({ embeds: [embed] });
+
+        // Update the last_message_id variable with the new message's ID
+        last_message_id = message.id;
+
+        // Store the new message ID in local storage
+        await persist.setItem(storage_key, { id: message.id });
+    } catch (error) {
+        console.error("Error while updating the door status message:", error);
+    }
+}
+
+function createDoorStatusEmbed(state: boolean): EmbedBuilder {
```

Comment:
Rename state to door_is_open

Reply:
https://github.com/GryphonRacingFSAE/Discord-Bot/pull/20/commits/9b2efb63f66e6bd18309bc97f0df7eb2c4700b60